File: android-app/xnative/src/main/java/uk/co/thetimes/xnative/ComponentsPackage.java
package uk.co.thetimes.xnative;

import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class ComponentsPackage implements ReactPackage {
    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        List<NativeModule> modules = new ArrayList<>();
        // TODO: Replace this with a library once react-native-text-size is updated: (https://github.com/aMarCruz/react-native-text-size)
        modules.add(new RNTextSizeModule(reactContext));
        return modules;
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }
}

File: android/app/src/main/java/com/storybooknative/ComponentsPackage.kt
package com.storybooknative

import android.view.View
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ReactShadowNode
import com.facebook.react.uimanager.ViewManager

class ComponentsPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext) = listOf(
            RNTextSizeModule(reactContext)
    )

    override fun createViewManagers(reactContext: ReactApplicationContext) = emptyList<ViewManager<View, ReactShadowNode<*>>>()
}
